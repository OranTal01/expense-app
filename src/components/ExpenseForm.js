import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import {history} from '../routers/AppRouter';


export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      amount: props.expense ? (props.expense.amount / 100).toString() : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: '',
      payments: '1'
    };
  }
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onAmountChange = (e) => {
    const amount = e.target.value;

    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };
  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({ error: 'Please provide description and amount.' }));
    }
    else if (this.state.payments > 1) {
      const userPayments = this.state.payments
      this.setState(() => ({ error: '' }));
      for (let index = userPayments; index >= 1; index--) {
        this.props.onSubmit({
          description: this.state.description,
          amount: (parseFloat(this.state.amount, 10) * 100) / (parseFloat(userPayments,10)),
          createdAt: index === userPayments ? this.state.createdAt.valueOf() : this.state.createdAt.add(1, 'months').valueOf(),
          note: this.state.note
        });
      }
    }
    else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  toggleAddEditExpense = () => {
    if (history.location.pathname === '/create') {
      return true
    } else {
      return false
    }
  };

  onChangePayments = (e) => {
    const userPayments = e.target.value;
    this.setState(() => ({
      payments: userPayments
    }));
  };

  render() {
    return (
      <form
      className="form"
        onSubmit={ this.onSubmit }>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
      <input
      className="text-input"
      type="text"
      placeholder="Description"
      autoFocus
      value={this.state.description}
      onChange={this.onDescriptionChange}
  />
  <input
    className="text-input"
    type="text"
    placeholder="Amount"
    value={this.state.amount}
    onChange={this.onAmountChange}
  />
  <SingleDatePicker
    date={this.state.createdAt}
    onDateChange={this.onDateChange}
    focused={this.state.calendarFocused}
    onFocusChange={this.onFocusChange}
    numberOfMonths={1}
    isOutsideRange={() => false}
  />
  <textarea
    className="textarea"
    placeholder="Add a note for your expense (optional)"
    value={this.state.note}
    onChange={this.onNoteChange}
  >
  </textarea>
  <div>
    <select
        className="select"
        value={this.state.payments}
        onChange={ this.onChangePayments }>
        <option value="1">Payments: 1</option>
        <option value="2">Payments: 2</option>
        <option value="3">Payments: 3</option>
        <option value="4">Payments: 4</option>
        <option value="5">Payments: 5</option>
        <option value="6">Payments: 6</option>
        <option value="7">Payments: 7</option>
        <option value="8">Payments: 8</option>
        <option value="9">Payments: 9</option>
        <option value="10">Payments: 10</option>
        <option value="11">Payments: 11</option>
        <option value="12">Payments: 12</option>
        <option value="13">Payments: 13</option>
        <option value="14">Payments: 14</option>
        <option value="15">Payments: 15</option>
        <option value="16">Payments: 16</option>
        <option value="17">Payments: 17</option>
        <option value="18">Payments: 18</option>
        <option value="19">Payments: 19</option>
        <option value="20">Payments: 20</option>
        <option value="21">Payments: 21</option>          
        <option value="22">Payments: 22</option>
        <option value="23">Payments: 23</option>
        <option value="24">Payments: 24</option>
      </select>
      </div>
      <div>
        { this.toggleAddEditExpense()
        ?
        <button className="button">Save Expense</button>
        :
        <button className="button">Save Edit Expense</button> }
      </div>
      </form>      
    )
  }
}
