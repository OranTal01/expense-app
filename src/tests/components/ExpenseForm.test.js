import React from 'react';
import { shallow } from 'enzyme';
import ExpenseForm from '../../components/ExpenseForm';
import moment from 'moment';
import expense from '../fixtures/expenses';


test('should render expense form', () => {
    const wrapper = shallow(<ExpenseForm />)
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseForm with expense data', () => {
    const wrapper = shallow(<ExpenseForm expense={ expense[0] } />)
    expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid form submission', () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    });
    expect(wrapper.state('error').length).toBeGreaterThan(0);
    expect(wrapper).toMatchSnapshot()
});

test('should set description on input change', () => {
    const value = 'description test'
    const wrapper = shallow(<ExpenseForm />)
    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('description')).toBe(value)
});

test('should set note on text arry change', () => {
    const value = 'note test'
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('textarea').simulate('change', {
        target: { value }
    });
    expect(wrapper.state('note')).toBe(value)
});

test('should set amount if valid', () => {
    const value = '30.50'
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('amount')).toBe(value)
});

test('should set amount if not valid', () => {
    const value = '30.505'
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('amount')).toBe('')
});

test('should set new date on date change', () => {
    const now = moment();
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('withStyles(SingleDatePicker)').prop('onDateChange')(now);
    ;
    expect(wrapper.state('createdAt')).toEqual(now);
});

test('should set focused on focuses change', () => {
    const focused = true
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('withStyles(SingleDatePicker)').prop('onFocusChange')({ focused });
    expect(wrapper.state('focused')).toEqual(true)
});
