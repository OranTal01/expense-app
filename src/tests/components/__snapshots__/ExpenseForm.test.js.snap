// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value=""
  />
  <SingleDatePicker
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    orientation="horizontal"
    phrases={
      Object {
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
        select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <select
      className="select"
      onChange={[Function]}
      value="1"
    >
      <option
        value="1"
      >
        Payments: 1
      </option>
      <option
        value="2"
      >
        Payments: 2
      </option>
      <option
        value="3"
      >
        Payments: 3
      </option>
      <option
        value="4"
      >
        Payments: 4
      </option>
      <option
        value="5"
      >
        Payments: 5
      </option>
      <option
        value="6"
      >
        Payments: 6
      </option>
      <option
        value="7"
      >
        Payments: 7
      </option>
      <option
        value="8"
      >
        Payments: 8
      </option>
      <option
        value="9"
      >
        Payments: 9
      </option>
      <option
        value="10"
      >
        Payments: 10
      </option>
      <option
        value="11"
      >
        Payments: 11
      </option>
      <option
        value="12"
      >
        Payments: 12
      </option>
      <option
        value="13"
      >
        Payments: 13
      </option>
      <option
        value="14"
      >
        Payments: 14
      </option>
      <option
        value="15"
      >
        Payments: 15
      </option>
      <option
        value="16"
      >
        Payments: 16
      </option>
      <option
        value="17"
      >
        Payments: 17
      </option>
      <option
        value="18"
      >
        Payments: 18
      </option>
      <option
        value="19"
      >
        Payments: 19
      </option>
      <option
        value="20"
      >
        Payments: 20
      </option>
      <option
        value="21"
      >
        Payments: 21
      </option>
      <option
        value="22"
      >
        Payments: 22
      </option>
      <option
        value="23"
      >
        Payments: 23
      </option>
      <option
        value="24"
      >
        Payments: 24
      </option>
    </select>
  </div>
  <div>
    <button
      className="button"
    >
      Save Edit Expense
    </button>
  </div>
</form>
`;

exports[`should render ExpenseForm correctly with expense data 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value="Rent"
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value="1095"
  />
  <SingleDatePicker
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1969-12-28T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    orientation="horizontal"
    phrases={
      Object {
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
        select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <select
      className="select"
      onChange={[Function]}
      value="1"
    >
      <option
        value="1"
      >
        Payments: 1
      </option>
      <option
        value="2"
      >
        Payments: 2
      </option>
      <option
        value="3"
      >
        Payments: 3
      </option>
      <option
        value="4"
      >
        Payments: 4
      </option>
      <option
        value="5"
      >
        Payments: 5
      </option>
      <option
        value="6"
      >
        Payments: 6
      </option>
      <option
        value="7"
      >
        Payments: 7
      </option>
      <option
        value="8"
      >
        Payments: 8
      </option>
      <option
        value="9"
      >
        Payments: 9
      </option>
      <option
        value="10"
      >
        Payments: 10
      </option>
      <option
        value="11"
      >
        Payments: 11
      </option>
      <option
        value="12"
      >
        Payments: 12
      </option>
      <option
        value="13"
      >
        Payments: 13
      </option>
      <option
        value="14"
      >
        Payments: 14
      </option>
      <option
        value="15"
      >
        Payments: 15
      </option>
      <option
        value="16"
      >
        Payments: 16
      </option>
      <option
        value="17"
      >
        Payments: 17
      </option>
      <option
        value="18"
      >
        Payments: 18
      </option>
      <option
        value="19"
      >
        Payments: 19
      </option>
      <option
        value="20"
      >
        Payments: 20
      </option>
      <option
        value="21"
      >
        Payments: 21
      </option>
      <option
        value="22"
      >
        Payments: 22
      </option>
      <option
        value="23"
      >
        Payments: 23
      </option>
      <option
        value="24"
      >
        Payments: 24
      </option>
    </select>
  </div>
  <div>
    <button
      className="button"
    >
      Save Edit Expense
    </button>
  </div>
</form>
`;

exports[`should render error for invalid form submission 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value=""
  />
  <SingleDatePicker
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    orientation="horizontal"
    phrases={
      Object {
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
        select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <select
      className="select"
      onChange={[Function]}
      value="1"
    >
      <option
        value="1"
      >
        Payments: 1
      </option>
      <option
        value="2"
      >
        Payments: 2
      </option>
      <option
        value="3"
      >
        Payments: 3
      </option>
      <option
        value="4"
      >
        Payments: 4
      </option>
      <option
        value="5"
      >
        Payments: 5
      </option>
      <option
        value="6"
      >
        Payments: 6
      </option>
      <option
        value="7"
      >
        Payments: 7
      </option>
      <option
        value="8"
      >
        Payments: 8
      </option>
      <option
        value="9"
      >
        Payments: 9
      </option>
      <option
        value="10"
      >
        Payments: 10
      </option>
      <option
        value="11"
      >
        Payments: 11
      </option>
      <option
        value="12"
      >
        Payments: 12
      </option>
      <option
        value="13"
      >
        Payments: 13
      </option>
      <option
        value="14"
      >
        Payments: 14
      </option>
      <option
        value="15"
      >
        Payments: 15
      </option>
      <option
        value="16"
      >
        Payments: 16
      </option>
      <option
        value="17"
      >
        Payments: 17
      </option>
      <option
        value="18"
      >
        Payments: 18
      </option>
      <option
        value="19"
      >
        Payments: 19
      </option>
      <option
        value="20"
      >
        Payments: 20
      </option>
      <option
        value="21"
      >
        Payments: 21
      </option>
      <option
        value="22"
      >
        Payments: 22
      </option>
      <option
        value="23"
      >
        Payments: 23
      </option>
      <option
        value="24"
      >
        Payments: 24
      </option>
    </select>
  </div>
  <div>
    <button
      className="button"
    >
      Save Edit Expense
    </button>
  </div>
</form>
`;

exports[`should render error for invalid form submission 2`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <p
    className="form__error"
  >
    Please provide description and amount.
  </p>
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Amount"
    type="text"
    value=""
  />
  <SingleDatePicker
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    orientation="horizontal"
    phrases={
      Object {
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
        select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a note for your expense (optional)"
    value=""
  />
  <div>
    <select
      className="select"
      onChange={[Function]}
      value="1"
    >
      <option
        value="1"
      >
        Payments: 1
      </option>
      <option
        value="2"
      >
        Payments: 2
      </option>
      <option
        value="3"
      >
        Payments: 3
      </option>
      <option
        value="4"
      >
        Payments: 4
      </option>
      <option
        value="5"
      >
        Payments: 5
      </option>
      <option
        value="6"
      >
        Payments: 6
      </option>
      <option
        value="7"
      >
        Payments: 7
      </option>
      <option
        value="8"
      >
        Payments: 8
      </option>
      <option
        value="9"
      >
        Payments: 9
      </option>
      <option
        value="10"
      >
        Payments: 10
      </option>
      <option
        value="11"
      >
        Payments: 11
      </option>
      <option
        value="12"
      >
        Payments: 12
      </option>
      <option
        value="13"
      >
        Payments: 13
      </option>
      <option
        value="14"
      >
        Payments: 14
      </option>
      <option
        value="15"
      >
        Payments: 15
      </option>
      <option
        value="16"
      >
        Payments: 16
      </option>
      <option
        value="17"
      >
        Payments: 17
      </option>
      <option
        value="18"
      >
        Payments: 18
      </option>
      <option
        value="19"
      >
        Payments: 19
      </option>
      <option
        value="20"
      >
        Payments: 20
      </option>
      <option
        value="21"
      >
        Payments: 21
      </option>
      <option
        value="22"
      >
        Payments: 22
      </option>
      <option
        value="23"
      >
        Payments: 23
      </option>
      <option
        value="24"
      >
        Payments: 24
      </option>
    </select>
  </div>
  <div>
    <button
      className="button"
    >
      Save Edit Expense
    </button>
  </div>
</form>
`;
